app/views/checkout/_form.html.haml

```diff
@@ -15,8 +15,8 @@
   = render "checkout/shipping", f: f
   = render "checkout/payment", f: f
   = render "checkout/already_ordered", f: f if show_bought_items?
-  = render "checkout/terms_and_conditions", f: f
+  = render_terms_and_conditions
   %p
-    %button.button.primary{ type: :submit, ng: { disabled: "terms_and_conditions_activated && !terms_and_conditions_accepted" } }
+    %button.button.primary{ type: :submit, ng: { disabled: "terms_and_conditions_accepted == false || platform_tos_accepted == false" } }
```

Comment:
but aren't `terms_and_conditions_accepted` and `platform_tos_accepted` booleans at this point? I see we initialize them like so.

Reply:
They are if the inputs are present. If TOS don't have to be accepted, the checkbox won't be there and `platform_tos_accepted` is `null`.